community/kernel/src/main/java/org/neo4j/kernel/impl/api/cursor/TxPropertyCursor.java

```diff
@@ -150,12 +152,47 @@ public boolean seek( int keyId )
     }
 
     @Override
-    public DefinedProperty getProperty()
+    public int propertyKeyId()
     {
-        if (property == null)
-            throw new IllegalStateException(  );
+        check();
+        return property.propertyKeyId();
+    }
+
+    @Override
+    public Object value()
+    {
+        check();
+        return property.value();
+    }
+
+    @Override
+    public boolean booleanValue()
+    {
+        return (Boolean) value();
+    }
+
+    @Override
+    public long longValue()
+    {
+        return ((Number) value()).longValue();
+    }
 
-        return property;
+    @Override
+    public double doubleValue()
+    {
+        return ((Number) value()).doubleValue();
+    }
+
+    @Override
+    public String stringValue()
+    {
+        return value().toString();
+    }
+
+    @Override
+    public void propertyData( WritableByteChannel channel )
+    {
+        throw new UnsupportedOperationException( "NYI" );
```

Comment:
Right now it is, yes. Writing property data is quite undefined at this point, so we need to revisit this once the new clients start using it.